src/autopas/options/OpenMPKindOption.h

```diff
@@ -0,0 +1,418 @@
+        {OpenMPKindOption::lb4omp_fac, "fac"},              // Factoring                    // Requires profiling
+        {OpenMPKindOption::lb4omp_faca, "faca"},            // Improved Factoring           // Requires profiling
+        {OpenMPKindOption::lb4omp_bold, "bold"},            // Bold                         // Requires profiling
+        {OpenMPKindOption::lb4omp_fac2, "fac2"},            // Practical Factoring
+        {OpenMPKindOption::lb4omp_wf, "wf"},                // Weighted Factoring
+        {OpenMPKindOption::lb4omp_af, "af"},                // Adaptive Factoring
+        {OpenMPKindOption::lb4omp_awf, "awf"},              // Adaptive Weighted Factoring
+        {OpenMPKindOption::lb4omp_tfss, "tfss"},            // Trapezoid Factoring Self Scheduling
+        {OpenMPKindOption::lb4omp_fiss, "fiss"},            // Fixed Increase Self Scheduling
+        {OpenMPKindOption::lb4omp_viss, "viss"},            // Variable Increase Self Scheduling
+        {OpenMPKindOption::lb4omp_rnd, "rnd"},              // Random
+
+        // LB4OMP's scheduling techniques used by Auto4OMP (in addition to the standard scheduling kinds):
+        {OpenMPKindOption::lb4omp_trapezoidal, "trapezoidal"},    // Trapezoid Self Scheduling
+        {OpenMPKindOption::lb4omp_fac2a, "fac2a"},                // Improved Practical Factoring
+        {OpenMPKindOption::lb4omp_static_steal, "static_steal"},  // Static with Steal enabled
+        {OpenMPKindOption::lb4omp_awf_b, "awf_b"},                // Adaptive Weighted Factoring Variant B
+        {OpenMPKindOption::lb4omp_awf_c, "awf_c"},                // Adaptive Weighted Factoring Variant C
+        {OpenMPKindOption::lb4omp_awf_d, "awf_d"},                // Adaptive Weighted Factoring Variant D
+        {OpenMPKindOption::lb4omp_awf_e, "awf_e"},                // Adaptive Weighted Factoring Variant E
+        {OpenMPKindOption::lb4omp_af_a, "af_a"},                  // Improved Adaptive Factoring
+#endif
+    };
+  };
+
+  /**
+   * Checks if a given kind is in a given list of kinds.
+   * @return whether the given kind is in the given list of kinds
+   */
+  template <typename... KindList>
+  static bool in(OpenMPKindOption k, KindList... list) {
+    return (... || (k == list));
+  }
+
+  /**
+   * Tells whether a given kind name is valid.
+   * @param name the name to test
+   * @return whether the name is a valid scheduling kind option
+   */
+  static bool valid(const std::string &name) {
+    return std::any_of(validNameAbbreviations.begin(), validNameAbbreviations.end(),
+                       [=](const std::string &abbreviation) { return contains(name, abbreviation); });
+  }
+
+  /**
+   * Parse the scheduling kind option from a string.
+   * @param the name of the scheduling kind
+   * @return the corresponding AutoPas option
+   */
+  static OpenMPKindOption parse(const std::string &name) {
```

Comment:
I wanted to also include the names used in the LB4OMP paper and GIT Readme, which differ a bit from their current versions. I used parse() to handle the name variations.

I used valid() in MD-Flexible's CLIParser, and only passed valid strings to parse(). This is because I had parse() return a default option for invalid strings (instead of an empty list or an error), which would go unnoticed by MD-Flexible. I'll adjust this to align with AutoPas's approach for the next request.